include/particle.tcc

```diff
@@ -456,18 +438,16 @@ bool mpm::Particle<Tdim, Tnphases>::update_volume_strainrate(unsigned phase,
 }
 
 // Compute mass of particle
-template <unsigned Tdim, unsigned Tnphases>
-bool mpm::Particle<Tdim, Tnphases>::compute_mass(unsigned phase) {
+template <unsigned Tdim>
+bool mpm::Particle<Tdim>::compute_mass() {
   bool status = true;
   try {
     // Check if particle volume is set and material ptr is valid
-    if (volume_(phase) != std::numeric_limits<double>::max() &&
-        material_.at(phase) != nullptr) {
+    if (volume_ != std::numeric_limits<double>::max() && material_ != nullptr) {
       // Mass = volume of particle * mass_density
-      this->mass_density_(phase) =
-          material_.at(phase)->template property<double>(
-              std::string("density"));
-      this->mass_(phase) = volume_(phase) * mass_density_(phase);
+      this->mass_density_ =
+          material_->template property<double>(std::string("density"));
+      this->mass_ = volume_ * mass_density_;
```

Comment:
If we want to treat the "particle" as a the solid phase used for all conditions(1p, 2p, 3p), should we consider to add a "volume fraction" and change the computation of mass as " this->mass_ = volume_ * mass_density_* volume_fraction_solid;"? And default it as 1 for "single phase".

Reply:
We can do that when we add `uvp` branch.